relay-filter/src/config.rs

```diff
@@ -386,4 +520,58 @@ mod tests {
         }
         "###);
     }
+
+    #[test]
+    fn test_deserialize_generic_filters() {
+        let json = r#"{
+            "version": 1,
+            "filters": {
+                "hydrationError": {
```

Comment:
I think moving the filter name to the filter configuration itself should have an easier to understand protocol, for example:

```json
{
  "version": 1,
  "filters": [
    {
        "id": "hydrationError",
        "enabled": true,
        "condition": { ... }
    }
  ]
}
```
It's also easier to deal with, as you can get a single object instead of a tuple. What do you think?

Reply:
The idea for the protocol, was to initially implement it very similarly to existing filters. For me it's not a problem to use the standard object implementation. @jjbayer what do you think?

Reply:
What I like about @iker-barriocanal's proposal is that it does not require custom (de)serializers, and that it preserves the order of filters regardless of the JSON implementation. I just read that the [JSON standard](https://www.json.org/) defines an object as an

> unordered set of name/value pairs.

In that light I would say let's implement @iker-barriocanal's proposal.

Reply:
Yep, this is why it was good, since we would remove the custom ser/deser code.